src/main/java/net/minecraftforge/common/brewing/IBrewingRecipe.java

```diff
@@ -20,26 +20,27 @@
 package net.minecraftforge.common.brewing;
 
 import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Recipe;
+import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraftforge.common.ForgeMod;
 
-public interface IBrewingRecipe {
+import java.util.function.Predicate;
 
-    /**
-     * Returns true is the passed ItemStack is an input for this recipe. "Input"
-     * being the item that goes in one of the three bottom slots of the brewing
-     * stand (e.g: water bottle)
-     */
-    boolean isInput(ItemStack input);
+public interface IBrewingRecipe extends Recipe<BrewingContainerWrapper>
+{
+    @Override
+    default boolean canCraftInDimensions(int p_43999_, int p_44000_)
+    {
+        return true;
+    }
 
-    /**
-     * Returns true if the passed ItemStack is an ingredient for this recipe.
-     * "Ingredient" being the item that goes in the top slot of the brewing
-     * stand (e.g: nether wart)
-     */
-    boolean isIngredient(ItemStack ingredient);
+    @Override
+    default RecipeType<?> getType()
+    {
+        return ForgeMod.BREWING;
+    }
 
-    /**
-     * Returns the output when the passed input is brewed with the passed
-     * ingredient. Empty if invalid input or ingredient.
-     */
-    ItemStack getOutput(ItemStack input, ItemStack ingredient);
+    Predicate<ItemStack> getReagent();
```

Comment:
A method returning a predicate feels awkward to me, wouldn't it be cleaner to just have an `is` method as before? Alternatively, do these still need to exist if `matches` does all the checking logic anyways? Would in theory let you have a recipe that matches depending on both the NBT of the base and a reagent if the two are separated.

If these are kept, a default implementation of `matches` would make sense as both your implementations have the same logic. I could also see these being kept in an abstract class but not in the base interface.

Reply:
well
1. it is a Predicate because there is no Ingredient that checks for a potion
2. they are needed for checking the validity of a reagent/base
3. yes the default implementation wold work
4. no it has to be in the base interface because it is checked for when checking item validity for reagent/base

Reply:
You did not answer my first question. What benefit is a predicate over `isBase(ItemStack)`? You already need an instance of `IBrewingRecipe`, so might as well just call a method on it over generating another lambda.

Reply:
I do with my next commit xD